src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,444 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-25, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+# pylint: skip-file
+
+from io import BytesIO
+from typing import Annotated, Dict, List, Optional
+import base64
+import copy
+import json
+import logging
+import time
+import uuid
+
+from fastapi import APIRouter, Request, Response
+from fastapi import Depends
+from fastapi import HTTPException
+from fastapi.responses import StreamingResponse
+from opentelemetry import trace
+from redis import RedisError
+
+from nv_ingest.framework.schemas.framework_message_wrapper_schema import MessageWrapper
+from nv_ingest.framework.util.service.impl.ingest.redis_ingest_service import RedisIngestService
+from nv_ingest.framework.util.service.meta.ingest.ingest_service_meta import IngestServiceMeta
+from nv_ingest_api.util.service_clients.client_base import FetchMode
+from nv_ingest_client.primitives.jobs.job_spec import JobSpec
+from nv_ingest_client.primitives.tasks.extract import ExtractTask
+
+# For PDF splitting
+import pypdfium2 as pdfium
+
+logger = logging.getLogger("uvicorn")
+tracer = trace.get_tracer(__name__)
+
+router = APIRouter()
+
+# Reuse V1 state constants and helper functions
+from ..v1.ingest import (
+    _get_ingest_service,
+    trace_id_to_uuid,
+    INGEST_SERVICE_T,
+    STATE_RETRIEVED_DESTRUCTIVE,
+    STATE_RETRIEVED_NON_DESTRUCTIVE,
+    STATE_RETRIEVED_CACHED,
+    STATE_FAILED,
+    STATE_PROCESSING,
+    STATE_SUBMITTED,
+    INTERMEDIATE_STATES,
+)
+
+
+def split_pdf_to_pages(pdf_content: bytes) -> List[bytes]:
```

Comment:
can we set this up to use logic, that allows the user to decide the split page interval (i.e. could be 1 but maybe I to break the pdf up into subsets of 10 pages).